node/pkg/p2p/cutover.go

```diff
@@ -0,0 +1,109 @@
+package p2p
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// The format of this time is very picky. Please use the exact format specified by cutOverFmtStr!
+const cutOverTimeStr = "2024-12-31T23:59:59-0000"
+const cutOverFmtStr = "2006-01-02T15:04:05-0700"
+
+// checkForCutOver checks to see if / when we need to cut over to the new quic-v1 based on a preset time. If cutOverTimeStr is set, then this feature is enabled.
+// If the current time is after the configured time, we should start up using the new quic-v1 bootstrap peers immediately. If the configured time is in the future,
+// we need to start up with the existing bootstrap string and cut over to the new quic-v1 at the specified time.
+func checkForCutOver(logger *zap.Logger, bootstrapPeers string, ccqBootstrapPeers string, components *Components) (newBootstrapPeers string, newCcqBootstrapPeers string, err error) {
+	newBootstrapPeers, newCcqBootstrapPeers, delay, err := checkForCutOverImpl(logger, bootstrapPeers, ccqBootstrapPeers, components, cutOverTimeStr, time.Now())
+
+	if delay != time.Duration(0) {
+		// Wait for the cut over time and then panic so we restart with the new quic-v1. TODO: Can we just restart p2p??
+		go func() {
+			time.Sleep(delay)
+			logger.Info("time to cut over to new quic-v1", zap.String("cutOverTime", cutOverTimeStr), zap.String("component", "p2pco"))
+			panic("p2pco: time to cut over to new quic-v1")
+		}()
+	}
+
+	return newBootstrapPeers, newCcqBootstrapPeers, err
+}
+
+func checkForCutOverImpl(logger *zap.Logger, bootstrapPeers string, ccqBootstrapPeers string, components *Components, cutOverTimeStr string, now time.Time) (newBootstrapPeers string, newCcqBootstrapPeers string, delay time.Duration, err error) {
```

Comment:
I renamed the local variable.

As far as having the `cutOverTimeStr` be a time.Time rather than a string, go does not allow const structs, so this seemed like the most straight forward way to do this. It allows someone to easily update the config time, run the tests to verify they have the format right, and commit. I would like to leave it as a string.